spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfigurationTests.java

```diff
@@ -543,6 +545,48 @@ void shouldCloseDiscoveryClient() throws Exception {
 		assertThat(isShutdown.get()).isTrue();
 	}
 
+	@Test
+	void httpComponentsProtocolUpgradeRequestCustomizer_shouldNotCreateCustomRequestCustomizer() {
+		setupContext(RefreshAutoConfiguration.class, AutoServiceRegistrationConfiguration.class);
+
+		Map<String, EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer> requestCustomizers = this.context
+				.getBeansOfType(EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer.class);
+
+		assertThat(requestCustomizers).hasSize(0);
+	}
+
+	@Test
+	void httpComponentsProtocolUpgradeRequestCustomizer_shouldEnableProtocolUpgrades() {
+		TestPropertyValues.of("eureka.client.http-components.enable-protocol-upgrades=true").applyTo(this.context);
+		setupContext(RefreshAutoConfiguration.class, AutoServiceRegistrationConfiguration.class);
+
+		assertThat(this.context.getBeansOfType(EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer.class))
+				.hasSize(1);
+		EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer httpComponentsProtocolUpgradeRequestCustomizer = this.context
+				.getBean("httpComponentsProtocolUpgradeRequestCustomizer",
+						EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer.class);
+
+		RequestConfig.Builder requestConfigBuilder = RequestConfig.custom().setProtocolUpgradeEnabled(false);
+		httpComponentsProtocolUpgradeRequestCustomizer.customize(requestConfigBuilder);
+		assertThat(requestConfigBuilder.build().isProtocolUpgradeEnabled()).isTrue();
+	}
+
+	@Test
+	void httpComponentsProtocolUpgradeRequestCustomizer_shouldDisableProtocolUpgrades() {
+		TestPropertyValues.of("eureka.client.http-components.enable-protocol-upgrades=false").applyTo(this.context);
+		setupContext(RefreshAutoConfiguration.class, AutoServiceRegistrationConfiguration.class);
+
+		assertThat(this.context.getBeansOfType(EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer.class))
+				.hasSize(1);
+		EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer httpComponentsProtocolUpgradeRequestCustomizer = this.context
+				.getBean("httpComponentsProtocolUpgradeRequestCustomizer",
+						EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer.class);
+
+		RequestConfig.Builder requestConfigBuilder = RequestConfig.custom().setProtocolUpgradeEnabled(true);
+		httpComponentsProtocolUpgradeRequestCustomizer.customize(requestConfigBuilder);
+		assertThat(requestConfigBuilder.build().isProtocolUpgradeEnabled()).isFalse();
+	}
```

Comment:
Please add a test to verify that user-added customizer will override property settings.